src/main/java/com/minecolonies/coremod/client/render/worldevent/WorldEventContext.java

```diff
@@ -20,27 +19,130 @@
  */
 public class WorldEventContext
 {
-    public static final WorldEventContext INSTANCE = new WorldEventContext();
+    private static WorldEventContext instance;
+
+    /**
+     * In chunks
+     */
+    private RenderLevelStageEvent stageEvent;
+    private int                   clientRenderDist;
+    private BufferSource          bufferSource;
+    private PoseStack             poseStack;
+    private float                 partialTicks;
+    private ClientLevel           clientLevel;
+    private LocalPlayer           clientPlayer;
+    private ItemStack             mainHandItem;
+    @Nullable
+    private IColonyView           nearestColony;
 
     private WorldEventContext()
     {
         // singleton
     }
 
-    RenderLevelStageEvent stageEvent;
-    BufferSource bufferSource;
-    PoseStack poseStack;
-    float partialTicks;
-    ClientLevel clientLevel;
-    LocalPlayer clientPlayer;
-    ItemStack mainHandItem;
-    @Nullable
-    IColonyView nearestColony;
+    /**
+     * Get the instance of the world event context.
+     *
+     * @return the world event context.
+     */
+    public static synchronized WorldEventContext getInstance()
```

Comment:
definitely not synchronized, elaborate why?

Reply:
Mostly to just shut Sonarlint up, but I can remove it xD